docs/learn/learn-staking-advanced.md

```diff
@@ -111,62 +111,75 @@ On Polkadot and Kusama, the instance of the pallet
+In {{ polkadot: Polkadot's :polkadot }}{{ kusama: Kusama's :kusama }} NPoS nomination intents are
+sorted in a list called [bags-list](https://github.com/paritytech/substrate/pull/9507).
+{{ kusama: The bags list example below uses DOT for explaining the concepts. :kusama }}The Bags-List
+substrate pallet is designed to be self-maintaining, with minimal effort from the blockchain, making
+it extremely scalable. This sorting functionality using bags is extremely important for the
 [long-term improvements](https://gist.github.com/kianenigma/aa835946455b9a3f167821b9d05ba376) of the
-staking/election system. The bags-list is capable of including an unlimited number of nodes, subject
-to the chain's runtime storage. In the current staking system configuration, the bags list keeps
+staking/election system.
+
+The bags list has two primary components, bags and nodes (or nominators' accounts), with bags
+containing the nodes with bonded balance within a specific range. In the figure below the 1st empty
+bag will contain nominators will take 21 - 30 DOT, the 2nd bag 11 - 20 DOT, and the 3rd bag 0-10
+DOT. The nomination intents are the nominators' accounts with bonded tokens (in this case they
+are 8) that will be put inside each of those three bags depending on the stake they have.
+
+![bags list example 0](../assets/bags-list-example-0.png)
+
+The bags list is semi-sorted, meaning that automatic sorting is only partially done. In fact, the
+nomination intents are automatically put into each bag based on the number of bonded tokens, but
+within each bag the nodes are arranged based on the time they are inserted and not by their stake.
+To sort nodes by stake within a bag there is a manual step (more on this later on). Note in the
+figure below how in the 2nd bag the node with 19 DOT (yellow circle) is the last one, meaning that
+the nomination intent was submitted after the other nodes. Same scenario for the node with 8 DOT
+(green circle) in the 3rd bag.
+
+The mentioned two nodes have the right to move up in their respective bags un put them in front of
+the nodes with less stake than them. This action of appropriately sorting the bag must be done
+manually by submitting the `putInFrontOf` extrinsic within the `voterList` pallet instance.
+Moreover, if for example the node with 19 DOT bonds an additional 2 DOT, that node will be put
+automatically in the 1st bag because the total number of bonded tokens will be outside the range of
+the 2nd bag (see figure below). That node with now 21 DOT will be put at the tail of the 1st bag
+with the possibility to manually put itself in front of "older" nodes with less than 21 DOT (if
+there are any).
+
+![bags list example 1](../assets/bags-list-example-1.png)
+
+If one decides to automatically send staking rewards to the stash account and bond them (i.e.
+compound staking), the position within a bag changes over time and this may result in a change of
```

Comment:
This is exactly the opposite. When the staking rewards are accumulated, this does not automatically change the node's position within the bag.

```suggestion
If one decides to send staking rewards to the stash account and automatically bond them (i.e.
compounding the staking rewards), the position within a bag does not change automatically. The same scenario applies to a slashing event, i.e., when a nominator gets slashed, their position within a bag does not change. This might result in a scenario where the node is in the wrong bag and needs to be placed in the right
```